src/ba-transport.c

```diff
@@ -685,11 +685,18 @@ static int transport_acquire_bt_a2dp(struct ba_transport *t) {
 	GUnixFDList *fd_list;
 	GError *err = NULL;
 
-	/* Check whether transport is already acquired - keep-alive mode. */
+	/* Check whether transport is already acquired - keep-alive mode.
+	 * We must ensure that transport release is not in progress before
+	 * testing the BT fd. Otherwise we might have the IO thread close it
+	 * after we have decided to reuse it! */
+	ba_transport_pthread_cleanup_lock(t);
```

Comment:
Thanks for nailing down the cause of race condition during PCM open. However, I think, that the lock should be done in the [bluealsa_pcm_open()](https://github.com/Arkq/bluez-alsa/blob/4c32fc30068a4d6faca462d8e7fbc64e3dec4c0b/src/bluealsa-dbus.c#L253), just before PCM fd is checked.
The D-Bus messages coming to the bluez-alsa are handled by a single thread, so there cannot be any race. So if I understand correctly, the PCM FIFO has been closes which caused IO thread to terminate itself. Before termination, the pcm.fd is checked (the lock in the IO thread), so the lock should be acquired also just before pcm.fd is checked in the D-Bus handler. Otherwise, during pcm open call someone might get busy error code which in fact is not true, because PCM is about to be freed.